pkg/oc/admin/diagnostics/diagnostics/cluster/aggregated_logging/diagnostic.go

```diff
@@ -177,22 +216,38 @@ func (d *AggregatedLogging) CanRun() (bool, error) {
 }
 
 func (d *AggregatedLogging) Check() types.DiagnosticResult {
-	if len(d.loggingURL) == 0 {
-		return d.result
+	if len(d.CmdlineProject) > 0 {
+		d.checkProjectDiagnostics(d.CmdlineProject)
+		return d.sumResult
 	}
 
-	project := retrieveLoggingProject(d.result, d.loggingURL, d.ProjectClient, d.RouteClient)
-	if len(project) != 0 {
-		checkServiceAccounts(d, d, project)
-		checkClusterRoleBindings(d, d, project)
-		checkSccs(d, d, project)
-		checkDeploymentConfigs(d, d, project)
-		checkDaemonSets(d, d, project)
-		checkServices(d, d, project)
-		checkRoutes(d, d, project)
-		checkKibana(d.result, d.RouteClient, d.OAuthClientClient, d.KubeClient, project)
+	for _, p := range loggingProjects {
+		d.checkProjectDiagnostics(p)
+		if d.projResult[p].Failure() {
+			d.Debug("AGL0020", fmt.Sprintf("Diagnostics for project '%s' have errors", p))
+		} else {
+			d.Debug("AGL0022", fmt.Sprintf("Diagnostics for project '%s' look ok", p))
+			return d.projResult[p]
```

Comment:
Why to return if one project has no errors? We should continue checks for all given loggingProjects

Reply:
AFAIK there can be only one aggregated logging project, checking each project will output confusing errors even when the correct project was found and validated

@jcantrill  can logging be deployed multiple times in multiple projects?

Reply:
logging can not be deployed in multiple namespaces.  The default is currently 'logging' with the new default to be 'openshift-logging' and unmodifable.  There is no need to take multiple namespaces as the stack would never function properly.  Fluentd is a daemonset, can only push to one logging stack and would fail the second deployment.